src/helpers/common.ts

```diff
@@ -197,3 +197,25 @@ export function isEmptyObject<T extends Record<any, unknown>>(
   }
   return true;
 }
+
+export async function asyncGeneratorToArray<T, L>(
+  generator: AsyncGenerator<T, L>,
+) {
+  const response = {
+    chunks: [] as T[],
+    output: undefined as L,
+  };
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
```

Comment:
Please avoid this pattern. I'd go with `while (!done)` in this case although a bit clunky.

Also, a good place for a unit test.